spatialdata/_core/transformations.py

```diff
@@ -197,10 +197,40 @@ def _get_n_spatial_dims(self, axes: tuple[str, ...]) -> int:
             raise ValueError(f"Invalid axes: {axes}")
         return valid_axes[axes]
 
-    def transform(self, element: SpatialElement) -> SpatialElement:
+    def transform(self, element: SpatialElement, maintain_positioning: bool = False) -> SpatialElement:
+        """
+        Transform a spatial element using this transformation and returns the transformed element.
+
+        Parameters
+        ----------
+        element
+            Spatial element to transform.
+        maintain_positioning
+            If True, each transformation of the transformed element will be prepended with the inverse of the current
+            transformation. In this way the data is transformed but the global positioning is maintained. An use case
+            changing the orientation/scale/etc of the data but keeping the alignment of the data within each coordinate
+            system. Please see notes for more details of how this parameter interact with xarray.DataArray for raster
+            data.
```

Comment:
can we come up with a better name of this? maybe `keep_alignment`?

Reply:
or maybe something like `update_transformations`? It would be helpful to explain what happens when set to False as well.

Reply:
Updated the docstring. Regarding the name @giovp I think that `keep_alignment` is slightly wrong because it's not a about the alignment of an element with respect to another but the positioning of the element in a coordinate system (this parameter may be set to true even with a spatialdata object with just one element). Anyway, I made the docstring cleared by replacing "global positioning" to "positioning in any coordinate system". I keep the name like this so we can merge, but feel free to open a issue about it or change it in a follow up PR if you find a better name.

Reply:
@kevinyamauchi please see above regarding the name `maintain_positioning`. I think the name `update_transformation` is also not working because the transformation is updated in any case when translating a raster data (since we don't translate the xarray coordinates), or when rotating an image. Anyway, I created now this issue for changing the name in the future. https://github.com/scverse/spatialdata/issues/148